include/TrustWalletCore/TWAnyAddress.h

```diff
@@ -43,6 +52,16 @@ bool TWAnyAddressIsValid(TWString* _Nonnull string, enum TWCoinType coin);
 TW_EXPORT_STATIC_METHOD
 struct TWAnyAddress* _Nullable TWAnyAddressCreateWithString(TWString* _Nonnull string, enum TWCoinType coin);
 
+/// Creates an bech32 address from a string representation, a coin type and the given hrp. Must be deleted with TWAnyAddressDelete after use.
+///
+/// \param string address to create.
+/// \param coin coin type of the address.
+/// \param hrp hrp of the address.
+/// \return TWAnyAddress pointer or nullptr if address and coin are invalid.
+TW_EXPORT_STATIC_METHOD
+struct TWAnyAddress* _Nullable TWAnyAddressCreateBech32(TWString* _Nonnull string, enum TWCoinType coin, TWString* _Nonnull hrp);
```

Comment:
Is this rename a good idea?  I think 'string' is better name here, as Address can be created from publickey or address string. The fact that it is Bech32 is a detail here.
I think there are quite a few developers, who are familiar with the chain they are using, but not necessary that the address format is Bec32.
I understand that addresses with HRP are Bech32 format, but I would not bet 100% even on this.

Reply:
> Is this rename a good idea? I think 'string' is better name here, as Address can be created from publickey or address string. The fact that it is Bech32 is a detail here. I think there are quite a few developers, who are familiar with the chain they are using, but not necessary that the address format is Bec32. I understand that addresses with HRP are Bech32 format, but I would not bet 100% even on this.

You are right HRP is a Bech32 format, my idea is to create clear separation with future format such as `SS58` (TWAnyAddressCreateSS58) for example - what do you think will be a good API in the future or naming?

Reply:
I suggested to make this change:
1. `WithStringAndHrp` is a bit redundant, you can know the string type from the argument, consider whole function signature user will see `bech32` and `hrp` 
2. With this change, we don't need to expose `TWBech32Address`
3. Similar naming convention for non default address methods, like `SS58` later

Reply:
My point is that if I want to validate a string-format address, I don't need to be aware that is Bech32/Base64/Hex or whatever.
AnyAddress is used generically, an address has its cointype and a string, that's all I need to know.
True, address schemes employing an HRP are Bech32, but can't we keep it as an internal detail?
Specialization to chain-specific implementation should be done inside the chain specific Entry or Address classes, not before calling AnyAddress.

Reply:
> 1. `WithStringAndHrp` is a bit redundant, you can know the string type from the argument, consider whole function signature user will see `bech32` and `hrp`

In C interface we have a lot of this kind of dependency, as polimorphism is not possible. I don't see problem with this.

Reply:
> My point is that if I want to validate a string-format address, I don't need to be aware that is Bech32/Base64/Hex or whatever. AnyAddress is used generically, an address has its cointype and a string, that's all I need to know. True, address schemes employing an HRP are Bech32, but can't we keep it as an internal detail? Specialization to chain-specific implementation should be done inside the chain specific Entry or Address classes, not before calling AnyAddress.

We could implement something like:

```cpp
TWAnyAddressCreateWithStringAndPrefix(TWString* _Nonnull string, enum TWCoinType coin, TWString* _Nonnull prefix);
```

Later, in Entry - end-chains know how to parse this kind of data:

- Cosmos -> HRP
- Polkadot -> Network prefix, convert the prefix to an integer.

Alternatively, it will be 

```cpp
TWAnyAddressCreateBech32(TWString* _Nonnull string, enum TWCoinType coin, TWString* _Nonnull hrp);
TWAnyAddressCreateSS58(TWString* _Nonnull string, enum TWCoinType coin, int prefix);
```

This API is less explicit but more flexible. What do you guys prefer @catenocrypt @hewigovens ?

Reply:
I like `TWAnyAddressCreateWithStringAndPrefix`. If it does not fit to a new unforseen use case in the future, it can be just duplicated at that time.

Reply:
You don't need to care about this by default if the coin is in registry.json. hrp or prefix exposes detail anyway and you need to document prefix + coin combinations now.

For developers want to use IBC chain, they probably know it already